core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -383,6 +383,14 @@ public MappingMetaData mapping(String mappingType) {
         return mappings.get(mappingType);
     }
 
+    public static final Setting<String> INDEX_MERGE_SOURCE_UUID = Setting.simpleString("index.merge.source.uuid");
```

Comment:
I wonder if we can find a different name than `merge` since it's somewhat overloaded?  `collapse` maybe?